server/src-lib/Hasura/GraphQL/Resolve/Action.hs

```diff
@@ -315,23 +319,25 @@ callWebhook manager reqHeaders confHeaders forwardClientHeaders resolvedWebhook
       J.toJSON e
     Right (Right responseWreq) -> do
       let responseValue = responseWreq ^. Wreq.responseBody
-      response <- decodeValue responseValue
-      case (_awrData response, _awrErrors response) of
-        (Nothing, Nothing) ->
-          throw500WithDetail "internal error" $
-          J.String "webhook response has neither 'data' nor 'errors'"
-        (Just _, Just _) ->
-          throw500WithDetail "internal error" $
-          J.String "webhook response cannot have both 'data' and 'errors'"
-        (Just d, Nothing) -> return d
-        (Nothing, Just errorResponse) -> do
-          let ActionWebhookError maybeCode maybeMessage = errorResponse
-              code = maybe Unexpected ActionWebhookCode maybeCode
-              withMessage message = err500 code message
-              noMessagekey = "\"message\" key is not found in webhook \"errors\" response"
-              withoutMessage = (err500 code noMessagekey)
-                               {qeInternal = Just $ J.object ["webhook_response" J..= responseValue]}
-          throwError $ maybe withoutMessage withMessage maybeMessage
+          responseStatus = responseWreq ^. Wreq.responseStatus
+
+      if | HTTP.statusIsSuccessful responseStatus  -> do
+             maybeData <- _awsData <$> decodeValue responseValue
+             onNothing maybeData $ throw500WithDetail "internal error" $
```

Comment:
this is going to be an interesting case where webhook returns 200 but hasura returns 500.

Reply:
Discussed with Vamshi, anything that breaks response contract is 500 for the client.